Source/Core/InputCommon/ImageOperations.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+
+namespace InputCommon
+{
+struct Pixel
+{
+  u8 r = 0;
+  u8 g = 0;
+  u8 b = 0;
+  u8 a = 0;
+
+  bool operator==(const Pixel& o) const { return r == o.r && g == o.g && b == o.b && a == o.a; }
+  bool operator!=(const Pixel& o) const { return !(o == *this); }
+};
+
+struct Point
+{
+  u32 x = 0;
+  u32 y = 0;
+};
+
+struct Rect
+{
+  Point p;
+  u32 width = 0;
+  u32 height = 0;
+};
```

Comment:
We also have a `Rectangle` class in MathUtil.
It uses top/left/right/bottom members though, so maybe you didn't want to work with that.

Reply:
Good call!  One disappointing thing is using `Rectangle<u32>` gives errors in VS because `std::abs` used in `GetHeight`/`GetWidth` is ambiguous.

Reply:
I should have fixed that recently. Do a rebase.

Reply:
In my local copy, I should be up-to-date with master.  Unfortunately, still the same error.

Reply:
I ended up keeping a similar signature to Rectangle with a note.